state/app/PreferenceManager.kt

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (C) 2022 Vaticle
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+package com.vaticle.typedb.studio.state.app
+
+import java.nio.file.FileSystems
+import java.nio.file.Path
+import kotlin.io.path.relativeTo
+
+class PreferenceManager(appData: DataManager) {
+    private val preferences = appData.preferences
+    val preferencesDialog = DialogManager.Base()
+
+    var autoSave: Boolean = Defaults.autoSave
+        get() = preferences.autoSave ?: field
+        set(value) = run { preferences.autoSave = value }
+
+    var graphOutputEnabled: Boolean = Defaults.graphOutputEnabled
+        get() = preferences.graphOutputEnabled ?: field
+        set(value) = run { preferences.graphOutputEnabled = value }
+
+    var matchQueryLimit: Long = Defaults.matchQueryLimit
+        get() = preferences.matchQueryLimit?.toLong() ?: field
+        set(value) = run { preferences.matchQueryLimit = value.toString() }
+
+    var ignoredPaths: List<String> = Defaults.ignoredPaths
+        get() = preferences.ignoredPaths ?: field
+        set(value) = run { preferences.ignoredPaths = value}
+
+    fun isIgnoredPath(path: Path): Boolean {
+        val ignoredPaths = preferences.ignoredPaths ?: Defaults.ignoredPaths
+        val relativePath = path.relativeTo(path.parent)
+        for (ignored in ignoredPaths) {
+            val pathMatcher = FileSystems.getDefault().getPathMatcher("glob:$ignored")
```

Comment:
I don't like this syntax:pattern submitted as a single string argument to `getPathMatcher`, but we get a ton of globbing functionality for free using a built-in Java library.

Reply:
Delighted that we get glob matching for our ignored paths! Let's add a caption that shows by example how to use it. Something like: `"e.g: *.md, data/**/out"`

Reply:
Done! Our caption is `"Supports wildcard matching, e.g.: .git, *.tql, README.?d, data/**/out"`